### PR TITLE
Various documentation fixes

### PR DESCRIPTION
**Pull request by wismill**
Typos, clarification and more examples.

By the way, should we add `doctest` to the CI?

### CODE REVIEW

**Comment by Bodigrim** [comment]
> By te way, should be add `doctest` to the CI?

Feel free to give it a try, but no guarantees that I'll maintain it working in a long run. My previous experience with doc tests was not great, so I'm likely to drop it as soon as it does not build / needs more setup.

File: src/Data/Text/Builder/Linear/Buffer.hs [hunk 1]
```diff
@@ -133,13 +133,13 @@ buffer |># addr# =
...
 --
--- The literal string must not contain zero bytes @\\0@ and must be a valid UTF-8,
+-- The literal string must not contain zero bytes @\\NUL@ and must be a valid UTF-8,
 -- these conditions are not checked.
 --
 -- /Note:/ When the syntactic extensions @UnboxedTuples@ or @UnboxedSums@ are
 -- enabled, extra spaces are required when using parentheses: i.e. use @( '#<|' )@
 -- instead of @('#<|')@. See the GHC User Guide chapter
--- “[Unboxed types and primitive operations](https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/primitives.html#unboxed-tuples)”
+-- “<https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/primitives.html#unboxed-tuples Unboxed types and primitive operations>”
```
Comment (Bodigrim) [inline src/Data/Text/Builder/Linear/Buffer.hs:142]: Any particular reason for this change?

File: src/Data/Text/Builder/Linear/Buffer.hs [hunk 2]
```diff
@@ -201,7 +201,7 @@ foldlIntoBuffer f = go
 
 -- $custom_hexadecimal
 --
--- Note that no /upper/ case hexadecimal formatting is provided. This package
--- provides a minimal API with utility functions only for common cases. For
--- other use cases, please adapt the code of this package, e.g. as shown in
--- the [Unicode code point example](https://github.com/Bodigrim/linear-builder/examples/src/Examples/Unicode.hs).
+-- Note that no /upper/ case nor padded hexadecimal formatting is provided. This
```
Comment (Bodigrim) [inline src/Data/Text/Builder/Linear/Buffer.hs:204]: I'm not a native speaker, but I'd expect "neither ... nor ...".

File: src/Data/Text/Builder/Linear/Hex.hs [hunk 3]
```diff
@@ -15,9 +15,10 @@ import GHC.ST (ST)
 
 import Data.Text.Builder.Linear.Core
 
--- | Append the lower-case hexadecimal represensation of a number.
+-- | Append the lower-case hexadecimal representation of a /bounded/ integral
+-- number.
 --
--- Negative numbers are interpreted as their corresponding unsigned number, e.g.
+-- Negative numbers are interpreted as their corresponding unsigned number, e.g.:
```
Comment (Bodigrim) [inline src/Data/Text/Builder/Linear/Hex.hs:21]: How about
```suggestion
-- Negative numbers are interpreted as their corresponding unsigned number:
```
and same below.

File: src/Data/Text/Builder/Linear/Buffer.hs (hunk 1 above)
Comment (wismill) [reply src/Data/Text/Builder/Linear/Buffer.hs:142]: For some reason it [is not parsed by haddock](https://hackage.haskell.org/package/text-builder-linear-0.1.2/docs/Data-Text-Builder-Linear-Buffer.html#v:-35--60--124-)

File: src/Data/Text/Builder/Linear/Buffer.hs (hunk 2 above)
Comment (wismill) [reply src/Data/Text/Builder/Linear/Buffer.hs]: Fixed

File: src/Data/Text/Builder/Linear/Hex.hs (hunk 3 above)
Comment (wismill) [reply src/Data/Text/Builder/Linear/Hex.hs]: Fixed

**Review by wismill — COMMENTED** [review]
Updated

**Review by Bodigrim — APPROVED** [review]

**Comment by Bodigrim** [comment]
Thanks!

**Pull request merged**